src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java

```diff
@@ -64,25 +64,32 @@
 
     private String bibtexEntryType;
 
-    // reduced subset, supports only "CITY , STATE, COUNTRY"
-    // \b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w*)\b
-    // WORD SPACE , SPACE WORD SPACE (zero or more) , SPACE WORD (zero or word)
-    // tested using http://www.regexpal.com/
-    //Matches both single locations like Berlin or Stroudsburg, PA, USA
-    private static final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w*)\\s?[,]?\\s?(\\w*)\\b");
-
-    // Allows 20.3-2007|||20/3-  2007 etc.
-    // (\d{1,2})\s?[.,-/]\s?(\d{1,2})\s?[.,-/]\s?(\d{2,4})
-    // 1-2 DIGITS SPACE SEPERATOR SPACE 1-2 DIGITS SPACE SEPERATOR SPACE 2-4 DIGITS
-    // tested using http://www.javaregex.com/test.html
+    /**
+     * reduced subset, supports only "CITY , STATE, COUNTRY" <br>
+     *  <b>\b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w*)\b</b> <br>
+     *  WORD SPACE , SPACE WORD SPACE (Can be zero or more) , SPACE WORD (Can be zero or more) <br>
+     *  Matches both single locations (only city) like Berlin and full locations like Stroudsburg, PA, USA <br>
+     *  tested using http://www.regexpal.com/
+     */
+    private final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w*)\\s?[,]?\\s?(\\w*)\\b");
+
+    /**
+     * Allows 20.3-2007|||20/3-  2007 etc.
+     * <b>(\d{1,2})\s?[.,-/]\s?(\d{1,2})\s?[.,-/]\s?(\d{2,4})</b>
+     * 1-2 DIGITS SPACE SEPERATOR SPACE 1-2 DIGITS SPACE SEPERATOR SPACE 2-4 DIGITS
+     */
     private static final Pattern DATE_PATTERN = Pattern
             .compile("(\\d{1,2})\\s*[.,-/]\\s*(\\d{1,2})\\s*[.,-/]\\s*(\\d{2,4})");
 
 
     public MSBibEntry() {
-
+        //empty
```

Comment:
Codacy and PMD has this as pattern ( UncommentedEmptyConstructor) to distinguish between internationally and unintentionally empty constructors
http://pmd.sourceforge.net/pmd-5.0.1/rules/java/design.html